internal/host/bannergrab.go

```diff
@@ -0,0 +1,166 @@
+package host
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	networkscan "github.com/Method-Security/networkscan/generated/go"
+	"github.com/praetorian-inc/fingerprintx/pkg/plugins"
+	"github.com/praetorian-inc/fingerprintx/pkg/scan"
+)
+
+// RunHostBannerGrab performs a banner grab on the specified target
+func RunHostBannerGrab(ctx context.Context, timeout int, target string) (*networkscan.BannerGrabReport, error) {
+	resources := networkscan.BannerGrabReport{Target: target}
+	errs := []string{}
+
+	host, portStr, err := net.SplitHostPort(target)
+	if err != nil {
+		return &resources, err
+	}
+
+	port, err := strconv.ParseUint(portStr, 10, 16)
+	if err != nil {
+		return &resources, err
+	}
+
+	ips, err := net.LookupIP(host)
```

Comment:
Why do we need to do this? FingerprintX can't work on FQDNs?

Reply:
The FingerprintX tool requires a port. Madhav and I talked about adding 443 or 80 if no port is sent in but decided against this since it is hidden from the user and imagined that if we have the address with the port it will inventible be sent in in the task run

Reply:
But the port and the fact that we're taking FQDNs and flipping over to IPs feel like they're unrelated unless I'm missing something. Ack that we need a port, that makes sense. I think we need that error handling higher up in the tool command then if it's so important.

Reply:
You need the FQDN for the Host field. I agree though the data that the tool requires is counter intuitive

> fxTarget := plugins.Target{
> Address: netip.AddrPortFrom(ipAddr, uint16(port)),
> Host:    host,
> }

ACK about the port, ill move the logic into the cmd file